stagecraft/apps/collectors/tests/test_views.py

```diff
@@ -0,0 +1,487 @@
+            "name": "some-provider",
+            "slug": "some-provider"
+        }
+
+        resp = self.client.post(
+            '/provider',
+            data=json.dumps(provider),
+            HTTP_AUTHORIZATION='Bearer development-oauth-access-token',
+            content_type='application/json')
+
+        assert_that(resp.status_code, equal_to(200))
+
+        resp_json = json.loads(resp.content)
+
+        assert_that(resp_json, has_key('id'))
+        assert_that(resp_json['name'], equal_to('some-provider'))
+        assert_that(resp_json['slug'], equal_to('some-provider'))
+
+    def test_post_from_unauthorised_client_fails(self):
+        provider = {
+            'name': 'some-provider',
+            "slug": "some-provider"
+        }
+        resp = self.client.post(
+            '/provider',
+            data=json.dumps(provider),
+            content_type='application/json')
+
+        assert_that(resp.status_code, equal_to(403))
+
+
+class DataSourceViewTestCase(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.provider = ProviderFactory()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.provider.delete()
+
+    def test_get(self):
+        data_source = DataSourceFactory()
+        response = self.client.get(
+            '/data-source/{}'.format(data_source.name),
+            HTTP_AUTHORIZATION='Bearer development-oauth-access-token',
+            content_type='application/json')
+
+        assert_that(response.status_code, equal_to(200))
+
+        resp_json = json.loads(response.content)
+
+        assert_that(resp_json['id'], equal_to(str(data_source.id)))
+        assert_that(resp_json['name'], equal_to(data_source.name))
+        assert_that(
+            resp_json['provider']['name'], equal_to(data_source.provider.name))
+        assert_that(resp_json, not(has_key('credentials')))
+
+    def test_get_from_unauthorised_client_fails(self):
+        data_source = DataSourceFactory()
+        resp = self.client.get(
+            '/data-source/{}'.format(data_source.name))
+
+        assert_that(resp.status_code, equal_to(403))
+
+    def test_post(self):
+        data_source = {
+            "name": "some-data-source",
+            "slug": "some-data-source",
+            "provider_id": self.provider.id
+        }
+
+        resp = self.client.post(
+            '/data-source',
+            data=to_json(data_source),
+            HTTP_AUTHORIZATION='Bearer development-oauth-access-token',
+            content_type='application/json')
+
+        assert_that(resp.status_code, equal_to(200))
+
+        resp_json = json.loads(resp.content)
+
+        assert_that(resp_json, has_key('id'))
+        assert_that(resp_json['name'], equal_to('some-data-source'))
+        assert_that(resp_json['slug'], equal_to('some-data-source'))
+        assert_that(resp_json['provider']['name'], equal_to(
+            self.provider.name))
+
+    def test_post_from_unauthorised_client_fails(self):
```

Comment:
I suppose maybe also the same for viewing data sources? That's how the ownership array works IIRC?